packages/migrate/src/transforms/2790-fix-transform.ts

```diff
@@ -0,0 +1,207 @@
+import ts from 'typescript';
+
+import path from 'path';
+
+import FixTransform, { FixedFile } from '../interfaces/fix-transform';
+import { findNodeAtPosition, insertStringIntoSourceFile } from '../helpers/typescript-ast';
+import RehearsalService from 'src/rehearsal-service';
+
+const QUESTION_MARK = '?';
+
+function findLocalInterface(
+  sourceFile: ts.SourceFile,
+  typeName: string
+): ts.InterfaceDeclaration | undefined {
+  const matched = sourceFile.statements.find(
+    (s) => ts.isInterfaceDeclaration(s) && s.name.getFullText().trim() === typeName
+  );
+  return matched as ts.InterfaceDeclaration;
+}
+
+function findImport(sourceFile: ts.SourceFile, typeName: string): ts.ImportDeclaration | undefined {
+  const matched = sourceFile.statements.find(
+    (s) =>
+      ts.isImportDeclaration(s) &&
+      (s.importClause?.namedBindings as ts.NamedImports).elements.find(
+        (e) => e.name.getFullText().trim() === typeName
+      )
+  );
+  return matched as ts.ImportDeclaration;
+}
+
+function findLocalClass(
+  sourceFile: ts.SourceFile,
+  typeName: string
+): ts.ClassDeclaration | undefined {
+  const matched = sourceFile.statements.find(
+    (s) => ts.isClassDeclaration(s) && s.name?.getFullText().trim() === typeName
+  );
+  return matched as ts.ClassDeclaration;
+}
+
+function updateTextWithLocalInterface(
+  sourceFile: ts.SourceFile,
+  declaration: ts.InterfaceDeclaration,
+  typeMemberName: string
+): FixedFile[] {
+  const matchedMember = (declaration as ts.InterfaceDeclaration).members.find(
+    (member) => member.name?.getFullText().trim() === typeMemberName
+  );
+  if (matchedMember) {
+    const nameEnd = (matchedMember as ts.PropertySignature).name.getEnd();
+    const updatedText = insertStringIntoSourceFile(sourceFile, nameEnd, QUESTION_MARK);
+    return [
+      {
+        fileName: sourceFile.fileName,
+        text: updatedText,
+      },
+    ];
+  }
+
+  return [];
+}
+
+function getSourceFileFromImport(
+  declaration: ts.ImportDeclaration,
+  basePath: string,
+  service: RehearsalService
+): ts.SourceFile {
+  const rawPath = declaration.moduleSpecifier.getFullText().trim();
+  const parts = rawPath
+    .replace(/^'(\.\/)?/, '') //remove quotation mark, and './' at the start
```

Comment:
You don't need to remove '.' or '..' and split string. The [resolve](https://nodejs.org/api/path.html#pathresolvepaths) takes care of them.

This will work just fine:

```typescript
resolve(basePath, `relativeModulePath${.ts}`);
```

but seems this solution won't work perfectly [because](https://www.typescriptlang.org/docs/handbook/module-resolution.html) and there are different methods of resolving modules:

```
At this point, the compiler will ask “what’s the shape of moduleA?” While this sounds straightforward, moduleA could be defined in one of your own **.ts/.tsx** files, or in a **.d.ts** that your code depends on.
```
I think the right solution is using ts.resolveModuleName... 
You will probably don't need to use basePath in this case as well...
Let me create a PR with moduleResolution quickly... I played with it recently.

Reply:
Changed to service.getResolveModuleName()